sqlmesh/core/loader.py

```diff
@@ -41,10 +41,10 @@
 
 # TODO: consider moving this to context
 def update_model_schemas(
-    dag: DAG[str], models: UniqueKeyDict[str, Model], context_path: Path
+    dag: DAG[str], models: UniqueKeyDict[str, Model], context: Context
```

Comment:
I'm against this as well

Reply:
Why can't we just pass a map from model names to model defaults?

Reply:
Or better yet a map from model name to a pair of default schema / catalog

Reply:
it's a hassle to build this